lib/mysql/index.js

```diff
@@ -127,21 +127,85 @@ const createEntities = (session) => {
     // schema name
     const schemaName = mysql.escape(session.schema);
 
-    // all the tables we need to look after
-    // session.entities is an array of strings that we transform into a single string
-    // where quotation marks are escaped using mysql.escape()
-    const tables = mysql.escape(session.entities);
+    // the array of columns we look after
+    const columns = mysql.escape(session.selectedColumns);
 
     // the query we pass
-    const query = queries.columns(schemaName, tables);
+    const query = queries.columns(schemaName, columns);
 
     return promisedQuery(query);
 };
 
 
+/*
+ * use the raw query results of mysqljs to create an object representing the database structure
+ * this function returns an object with as many properties as there are tables
+ * each table is an array of objects, each one of these objects represents a column
+ *
+ * @param { Object } queryResults - the output of a mysql query
+ * @param { string[] } tables - the selected tables
+ * @returns { Object } the returned object has this structure : { table_name: { column_name: { other query results } } }
+ */
+const organizeColumns = (queryResults, tables) => {
+    const result = {};
+
+    // modify this loop to add more column values
+    queryResults.forEach((queryResult, index) => {
+        const tableName = queryResult.table_name;
+        const columnName = queryResult.column_name;
+        const ordinalPosition = queryResult.ordinal_position;
```

Comment:
I keep the ordinal_position in order to generate the fields in the same order the dbms would display them. It's easier to compare the result JSON file and the table like this. However we don't need to create the jhipster properties fieldId (and relationshipId ?) indeed because they are legacy.